Core/clim-core/dialog-views.lisp

```diff
@@ -101,6 +101,85 @@
                                     :value default
                                     :value-changed-callback #'value-changed-callback))))
 
+;;; radio-box-view
+
+(define-presentation-method accept-present-default
+    ((type completion) stream (view radio-box-view) default default-supplied-p
+     present-p query-identifier)
+  (updating-output (stream
+                    :cache-value t      ; don't redisplay
+                    :unique-id query-identifier
+                    :record-type 'accepting-values-record)
+    (with-look-and-feel-realization ((frame-manager *application-frame*) *application-frame*)
+      (with-output-as-gadget (stream)
+        (let* ((buttons (loop for choice in sequence collect
+                             (make-pane 'toggle-button
+                                        :indicator-type :one-of
+                                        :id (funcall value-key choice)
+                                        :label (funcall name-key choice))))
+               (initargs (loop for slot in (c2mop:class-slots (find-class 'radio-box-view))
+                            when (slot-boundp view (c2mop:slot-definition-name slot)) append
+                              (list (car (c2mop:slot-definition-initargs slot)) (slot-value view (c2mop:slot-definition-name slot)))))
+               (gadget (apply #'make-pane 'radio-box :choices buttons
+                              :current-selection (find default buttons :test test :key #'gadget-id)
+                              :value-changed-callback
+                              (lambda (pane item)
+                                (let ((value (gadget-id item)))
+                                  (throw-object-ptype `(com-change-query ,query-identifier ,value) '(command :command-table accept-values))))
+                              initargs)))
+          gadget)))))
+
+;;; check-box-view
+
+(define-presentation-method accept-present-default
```

Comment:
Please extract the identical parts of this method and the previous method into a helper function.

Reply:
I think now it is better.